analysis/analyze.js

```diff
@@ -0,0 +1,183 @@
+const fs = require('fs');
+
+const yargs = require('yargs');
+const { BigQuery } = require('@google-cloud/bigquery');
+
+async function sleep(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function getOrCreateDataset(bigquery, datasetName) {
+  console.log(`Ensuring dataset ${bigquery.projectId}.${datasetName} exists.`);
+
+  const [dataset] = await bigquery
+    .dataset(datasetName, {
+      location: 'US'
+    })
+    .get({
+      autoCreate: true
+    });
+
+  return dataset;
+}
+
+async function createTableFromQuery(bigquery, dataset, query) {
+  const sql = query.sql;
+  const destination = dataset.table(query.destination);
+
+  console.log(`Creating table ${bigquery.projectId}.${dataset.id}.${destination.id} from query.`);
+
+  try {
+    const [job] = await bigquery.createQueryJob({
+      query: sql,
+      destination,
+      location: 'US'
+      // useQueryCache: false,
```

Comment:
can we remove these lines?

Reply:
This one yes, I'd like to keep the second comment which is useful when you simply want to repopulate all your tables without removing them before.

I had a hard time finding that option in the BigQuery node client library docs and would love to keep it in the code for reference (it's missing in https://googleapis.dev/nodejs/bigquery/latest/BigQuery.html#createQueryJob and you have to dig into https://cloud.google.com/bigquery/docs/reference/rest/v2/Job#JobConfigurationQuery to find these options)…